pkg/controller/machine.go

```diff
@@ -372,49 +372,53 @@ func (c *Controller) syncHandler(key string) error {
 
 // deleteMachineAndProviderInstance makes sure that an instance has gone in a series of steps.
 func (c *Controller) deleteMachineAndProviderInstance(prov cloud.Provider, machine *machinev1alpha1.Machine) error {
-	var err error
-	var providerInstance instance.Instance
 
-	// step 1: delete provider instance
-	if err = c.deleteProviderInstance(prov, machine); err != nil {
-		message := fmt.Sprintf("%v. Please manually delete finalizers from the machine object.", err)
-		return c.updateMachineErrorIfTerminalError(machine, machinev1alpha1.DeleteMachineError, message, err, "failed to delete machine at cloudprovider")
-	}
+	// step 1: get the provider instance.
+	providerInstance, err := c.getProviderInstance(prov, machine)
+	if err != nil {
+		// step 1.1: failed to get instance, because of some unknown error -> return and see if we need to handle a terminal error here
+		if err != cloudprovidererrors.ErrInstanceNotFound {
+			return c.updateMachineErrorIfTerminalError(machine, machinev1alpha1.DeleteMachineError, err.Error(), err, fmt.Sprintf("failed to get instance for machine %s after the delete got triggered", machine.Name))
+		}
 
-	// step 2: check that the instance has really gone
-	if providerInstance, err = c.getProviderInstance(prov, machine); err == nil {
-		if sets.NewString(string(instance.StatusDeleted), string(instance.StatusDeleting)).Has(string(providerInstance.Status())) {
-			glog.V(4).Infof("deletion of instance %s got triggered. Waiting until it fully disappears", providerInstance.ID())
-			c.workqueue.AddAfter(machine.Name, deletionRetryWaitPeriod)
-			return nil
+		// step 1.2 the instance could not be found -> it's gone, so we remove the finalizer. This essentially will remove the machine object from the system
+		glog.V(4).Infof("Deleted machine %s at cloud provider", machine.Name)
```

Comment:
Can we try to use max level 3 for our logging because at 4 is the output of the leader election which is extremely verbose?

Reply:
4 is default. I would like to not bend our code for the fact that the leaderelection log level is poorly set